tests/basic/storage_pool_test.go

```diff
@@ -9894,3 +9898,219 @@ var _ = Describe("{AddDriveWithKernelPanic}", func() {
+				log.FailOnError(err, "error in getting pool with IO with error")
+				if drvs, ok := drvMap[fmt.Sprintf("%d", poolListForOps[i].ID)]; ok {
+					maxDrivesAllowed = POOL_MAX_CLOUD_DRIVES - len(drvs)
+					numofDrivesInNode := GetNumOfDrivesInNode(selectedNode)
+					//check if drives added  to pool will more than permissible for the node
+					if (int(maxDriveLimit) - numofDrivesInNode) > maxDrivesAllowed {
+						selectedPool = poolListForOps[i]
+						break
+					}
+				}
+			}
+			isjournal, err := IsJournalEnabled()
+			log.FailOnError(err, "Failed to check is journal enabled")
+			for i := 1; i <= maxDrivesAllowed+1; i++ {
+				drvSize, err := getPoolDiskSize(selectedPool)
+				log.FailOnError(err, "failed while getting pool size")
+				driveSize := drvSize * uint64(i)
+				expectedSize := (selectedPool.TotalSize / units.GiB) + driveSize
+				err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+				log.FailOnError(err, "error while expanding pool")
+				resizeErr := waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isjournal)
+				if i == maxDrivesAllowed+1 {
+					poolStatus, err := getPoolLastOperation(selectedPool.Uuid)
+					log.FailOnError(err, "error getting pool status")
+					dash.VerifyFatal(resizeErr != nil, true, fmt.Sprintf("resize successful on pool %s", selectedPool.Uuid))
+					if poolStatus != nil {
+						dash.VerifyFatal(strings.Contains(poolStatus.Msg, fmt.Sprintf("unable to add 1 new drive(s) as it would exceed maximum supported drives (%v)", POOL_MAX_CLOUD_DRIVES)), true, "Error expected as drive added more than allowed per pool")
+					} else {
+						log.FailOnError(fmt.Errorf("failed while getting msg"), "failed while getting msg")
+					}
+				} else {
+					dash.VerifyFatal(resizeErr != nil, false, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+				}
+			}
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+		})
+		stepStr = "Test beyond max pool count"
+		Step(stepStr, func() {
+			log.InfoD(stepStr)
+			//search for suitable node
+			stNodes := node.GetStorageNodes()
+			dash.VerifyFatal(len(stNodes) > 0, true, "Storage nodes found?")
+			for _, node := range stNodes {
+				poolList, err := GetPoolsDetailsOnNode(node)
+				log.FailOnError(err, "failed while getting updated node pool list")
+				poolLeft := maxPoolLength - len(poolList)
+				numofDrivesInNode := GetNumOfDrivesInNode(selectedNode)
+				if poolLeft < int(maxDriveLimit)-numofDrivesInNode {
+					selectedNode = node
+					break
+				}
+			}
+			//add new pools to selected node
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			log.FailOnError(err, "failed while getting updated node pool list")
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
+			log.FailOnError(err, "error adding new pools")
+			err = addNewPools(selectedNode, 1)
```

Comment:
why addNewPools with PoolLeft and 1 ?  can you please relevant comment ?

Reply:
this is added to get all new pool added till supported and then work with existing pool